src/crepe/api/Button.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <functional>
+#include "UiObject.h"
+
+namespace crepe {
+
+/**
+ * \class Button
+ * \brief Represents a clickable UI button, derived from the UiObject class.
+ * 
+ * This class provides functionality for a button in the UI, including toggle state,
+ * click handling, and mouse hover detection. A callback function can be provided to
+ * handle button clicks.
+ */
+class Button : public UiObject {
+public:
+	/**
+     * \brief Constructs a Button with the specified game object ID and dimensions.
+     * 
+     * \param id The unique ID of the game object associated with this button.
+     * \param width The width of the button.
+     * \param height The height of the button.
+     * \param is_toggle Optional flag to indicate if the button is a toggle button. Defaults to false.
+     * \param on_click callback function that will be invoked when the button is clicked.
+     */
+	Button(game_object_id_t id, int width, int height, bool is_toggle = false, std::function<void()> on_click = nullptr);
+
+	/**
+     * \brief Indicates if the button is a toggle button (can be pressed and released).
+     * 
+     * A toggle button allows for a pressed/released state, whereas a regular button
+     * typically only has an on-click state.
+     */
+	bool is_toggle;
+
+	/**
+     * \brief Indicates whether the button is currently pressed.
+     * 
+     * This state is true when the button is actively pressed and false otherwise.
+     */
+	bool is_pressed;
+
+	/**
+     * \brief Indicates whether the mouse is currently hovering over the button.
+     * 
+     * This is set to true when the mouse is over the button and false otherwise.
+     */
+	bool hover;
```

Comment:
Ik zie dat je de default waardes hier weg hebt gehaald. Zo lang je member variabele niet set d.m.v. de constructor (wat het geval is bij hover en is_pressed), is het de bedoeling dat je default waardes gewoon hier (in de .h file) aangeeft.